src/components/productrecommendations.main.jsx

```diff
@@ -38,27 +38,113 @@ class ProductRecommendationsDisplayMain extends React.Component {
+                }
+        }
+    
+        return <div>{data}</div>;
+    }
+    renderCarouselView(currentItem, totalCount, product, MaxItemsInOneCarouselView) {
+        /* Renders Each View in Carousel*/
+        var some = MaxItemsInOneCarouselView;
+        var data = [];
+        if ( currentItem < totalCount  && product[currentItem].rel === "element"){
+            console.log("renderView: " + currentItem);
+            data.push(  
+                    <div className="col-md-6 col-sm-6 col-6" key ={'_' + Math.random().toString(36).substr(2, 9)}  >
+                        <ProductListItemMain productUrl={product[currentItem].uri} />
+                    </div>    
                 );
+        }
+        for (var i = 1 ; i<MaxItemsInOneCarouselView;i++){
+            /* Copy this section to add more items in the same carousel view*/
+            currentItem++;
+            if ((currentItem) < totalCount && product[(currentItem)].rel === "element"){
+                console.log("renderView: " + (currentItem));
+                data.push(
+                        <div className="col-md-6 col-sm-6 col-6" key = {'_' + Math.random().toString(36).substr(2, 9)} >
+                            <ProductListItemMain productUrl={product[(currentItem)].uri} />
+                        </div>    
+                    );
             }
-        })
+        }
+        
+        return data;
+
     }
     render() {
         if (this.state.productData.links.length > 0) {
             console.log(this.state.productData['_recommendations'][0]['_crosssell'][0])
             return (
                 <div data-region="categoryBrowseRegion" style={{ display: 'block' }}>
                     <label className="control-label">Product Recommendations</label>
-                    <ul className="category-items-listing equalize" id="category_items_listing">
-                        {this.renderProducts()}
-                    </ul>
+                    <div className="col-md-12">
+                        <div className="carousel slide" data-ride="carousel" id="theCarousel">
+                        <div className="container">
+                            <div className="carousel-inner">
+                            {this.renderRecommendations()}
+                            </div>
+                        </div> 
+                        <a className="left carousel-control" href="#theCarousel" data-slide="prev"><i className="glyphicon glyphicon-chevron-left"></i></a>
+                        <a className="right carousel-control" href="#theCarousel" data-slide="next"><i className="glyphicon glyphicon-chevron-right"></i></a>                           
+                        </div>
+                    </div>
+
+                    {/* Copy this section to add new carousel */}
```

Comment:
Is this your comment?

Reply:
yes, to make it easier for external devs to customize the code.